infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/PVCSubPathHelper.java

```diff
@@ -218,17 +225,23 @@ private void execute(
     KubernetesDeployments deployments = null;
     try {
       deployments = factory.access(workspaceId, namespace).deployments();
-      deployments.create(pod);
-      watchLogsIfDebugEnabled(deployments, pod, identity, startOptions);
-      final Pod finished = deployments.wait(podName, WAIT_POD_TIMEOUT_MIN, POD_PREDICATE::apply);
-      PodStatus finishedStatus = finished.getStatus();
-      if (POD_PHASE_FAILED.equals(finishedStatus.getPhase())) {
-        String logs = deployments.getPodLogs(podName);
-        LOG.error(
-            "Job command '{}' execution is failed. Logs: {}",
-            Arrays.toString(command),
-            Strings.nullToEmpty(logs).replace("\n", " \\n")); // Force logs onto one line
+
+      if (pvcExistsAndBound(namespace, pvcName)) {
```

Comment:
The PR also checks if PVC exists and is bound before creating the pod. This is because the PVC could have been in the processing of terminating before `execute()` is run.

Also it seems that because of this `if` condition, only 1/9 tests in PVCSubPathHelperTest.java succeed. I will update/add tests very soon.

Reply:
it's going to fail in WaitForFirstConsumer storage type.
it's default for RHPDS cluster and pretty common I think.